extensions/wikia/GDPR/UserGlobalProfileDataDelete.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+class UserGlobalProfileDataDelete {
+	public static function deleteGlobalProfileData( User $user ): bool {
+		if ( !( $user instanceof User ) || $user->getId() === 0 ) {
```

Comment:
* the class of `$user` argument is forced by the type hint, no need to check instance later on
* let's use `isAnon` instead of checking if user id is zero.